src/logging.rs

```diff
@@ -176,7 +184,14 @@ impl SealedLog for LogStorage {
                 return;
             }
         }
-        let message = record.args().to_string();
+        let mut message = record.args().to_string();
+
+        if let Some(max_line_length) = self.max_line_length {
+            if message.len() > max_line_length {
+                message = format!("{}...", &message[..max_line_length - 3]);
```

Comment:
This will panic if the length is not on a utf-8 character boundary, `floor_char_boundary` would be perfect here, but it's unstable still. I think the simplest approach without it would be

```rust
let mut length = max_line_length - 3;
while !message.is_char_boundary(length) {
  length -= 1;
}
message = format!("{}...", &message[..length]);
```

Reply:
@Nemo157 thanks for the hint! updated the code accordingly